nix/pkgs/mayastor/default.nix

```diff
@@ -22,114 +22,14 @@
 , targetPackages
 , buildPackages
 , targetPlatform
+, pkgs
 }:
 let
-  channel = import ../../lib/rust.nix { inherit sources; };
-  rustPlatform = makeRustPlatform {
-    rustc = channel.stable.rust;
-    cargo = channel.stable.cargo;
-  };
-  whitelistSource = src: allowedPrefixes:
-    builtins.filterSource
-      (path: type:
-        lib.any
-          (allowedPrefix:
-            lib.hasPrefix (toString (src + "/${allowedPrefix}")) path)
-          allowedPrefixes)
-      src;
   version = (builtins.fromTOML (builtins.readFile ../../../mayastor/Cargo.toml)).package.version;
-  src_list = [
-    "Cargo.lock"
-    "Cargo.toml"
-    "cli"
-    "csi"
-    "devinfo"
-    "jsonrpc"
-    "mayastor"
-    "nvmeadm"
-    "rpc"
-    "spdk-sys"
-    "sysfs"
-    "control-plane"
-    "composer"
-  ];
-  buildProps = rec {
-    name = "mayastor";
-    #cargoSha256 = "0000000000000000000000000000000000000000000000000000";
-    cargoSha256 = "0kzbbfg36rcly8xif64pqazf8687skwp3ajrlhf90d7g5rnc9lfn";
-    inherit version;
-    src = whitelistSource ../../../. src_list;
-    LIBCLANG_PATH = "${llvmPackages.libclang}/lib";
-    PROTOC = "${protobuf}/bin/protoc";
-    PROTOC_INCLUDE = "${protobuf}/include";
-
-    nativeBuildInputs = [
-      pkg-config
-      protobuf
-      llvmPackages_11.clang
-    ];
-    buildInputs = [
-      llvmPackages_11.libclang
-      protobuf
-      libaio
-      libiscsi.lib
-      libudev
-      liburing
-      numactl
-      openssl
-      utillinux
-    ];
-    verifyCargoDeps = false;
-    doCheck = false;
-    meta = { platforms = stdenv.lib.platforms.linux; };
-  };
+  project-builder = pkgs.callPackage ./cargo-package.nix { inherit version; };
```

Comment:
It would be perhaps easier to navigate if these were consistently named. 

You did some restructuring of these nix files to allow single binaries! I was looking at this too! I'm not sure this is the most idiomatic way, but I have no idea what that way is yet... I think this is a definite improvement though. :)

Reply:
Yeah that can be done too, my main focus here was to allow building just the mayastor binary only as part of the "units" to speed up the control plane builds as we just want the mayastor binary and nothing else.

To get the single binaries on mayastor it's probably better to build the whole project and then have derivations that expose each binary otherwise you might be rebuilding the common imports more than once.

Reply:
Also, I'm not sure if we want to keep the mayastor/mayastor-dev as the collection of all binaries, just the one binary or a few select. If we have the answer to that I'm happy to add that change to this PR.
I also don't know what's the most idiomatic in nix :/

Reply:
Hmm let's leave it for a future PR? :)